### PR TITLE
Start sphinx docs

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- #169 install docs started
- #165 as-is architecture started
  - will probably upstream this there so i can start documenting what the proposed changes might be

## Code changes

Goals for this PR:
- [ ] collect envs/reqs into a folder
  - [x] Adds new (nb)sphinx env/requirements
- [x] Adds doc script
- [x] Handle some initial documentation tasks
- [ ] icon
- [ ] review some best practices
  - [ ] https://www.divio.com/blog/documentation/
- [ ] Add docs to automation
  - [ ] integrity
  - [ ] linting
  - [ ] CI
- [ ] integrate existing content:
  - [ ] ROADMAP
  - [ ] README 
  - [ ] CONTRIBUTNG
  - [ ] LANGUAGESERVERS
  - [ ] EXTENDING

(Maybe) non-goals for this PR:
- [ ] add readthedocs
- [ ] add screenshots/gifs from acceptance tests

## User-facing changes

None, yet

## Backwards-incompatible changes

None, yet

## Chores

- [ ] linted
- [ ] tested
- [ ] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
Working on the docs locally with autoreload:
Screenshot from 2020-01-24 09-44-23

File: requirements-docs.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# development dependencies for jupyter_lsp for qa, with lab
```
Comment (krassowski) [inline requirements-docs.txt]: Could we please move the `requirements-*` `environment-*`, `setup-*` and other files to a subdirectory? We already have 42 files in the root. It is literally hard to scroll down to readme. This PR will bump it to 45.

Obviously I understand that not everything can be moved to a subdirectory due to some of the CI tools being stubborn, but I believe that we should strive for having no more than 15 files in the root. Happy to help with restructuring!

Also, `LANGUAGESERVERS.md` and `EXTENDING.md` could be moved to docs (not a strong opinion, just a suggestions).

Comment (bollwyvl) [reply requirements-docs.txt]: Happy to do some reorg: will start a separate PR and upstream it into this. However we get them into the docs, I'd move to keep at least the following in the root:
- README.md
- CONTRIBUTING.md

I don't know about travis, but we can move azure... but it's a project level thing, and should likely be done on another PR.

We can go back to `binder` folder, which would make that `environment.yml` the "demo" environment.

Comment (krassowski) [reply requirements-docs.txt]: Thank you! Agreed, `README.md` and `CONTRIBUTING.md` should definitely should in the root. `CHANGELOG.md` and some others  - maybe...

**Comment by bollwyvl** [comment]
This is mostly wired up now, and at least generates the docs on azure. I've also updated the architecture pieces a fair amount (i.e. documented #165 here, rather than on that branch), and spent a little more time on the install piece.  There are a number of outstanding issues, of course, but it might make sense to merge and move forward more incrementally: it's not ready to start publishing on readthedocs, and indeed, we may not _want_ to do so, if we want to use screenshots from the acceptance tests in the docs.

Some future, non-content thoughts:
- test the links with e.g. [pytest-check-links](https://pypi.org/project/pytest-check-links) 
- apply some more rigorous formatting and linting with e.g. [prenotebook](https://github.com/deathbeds/prenotebook)

File: docs/EXTENDING.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 > potential points of extension, but will require some refactoring to achieve.
 
-## jupyter-lsp
+### jupyter-lsp
 
-### Language Server Specs
+#### Language Server Specs
 
-Language Server Specs can be [configured](./LANGUAGESERVERS.ms) by Jupyter users,
+Language Server Specs can be [configured](./LANGUAGESERVERS.md) by Jupyter users,
```
Comment (krassowski) [inline docs/EXTENDING.md]: nice catch!

**Comment by krassowski** [comment]
Waiting for the CI to pass before I merge.

**Comment by krassowski** [comment]
The tests like to fail on master...

**Pull request merged**